Javadoc generation fails when using java 8

Java 8 is more strict when it comes to Javadoc (see http://openjdk.java.net/jeps/172). As a result you cannot generate javadoc (e.g. through the `mvn javadoc:javadoc` command) when using java 8.

@kaspersorensen I tried this on the master branch. For me `mvn javadoc:javadoc` fails with errors like these:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:2.9.1:javadoc (default-cli) on project DataCleaner-api: An error has occurred in Ja
vaDocs report generation:
[ERROR] Exit code: 1 - C:\git\DataCleaner\api\src\main\java\org\datacleaner\api\AnalyzerResultFuture.java:64: warning: no description for @param
[ERROR] * @param listener
[ERROR] ^
[ERROR] C:\git\DataCleaner\api\src\main\java\org\datacleaner\api\AnalyzerResultFuture.java:71: warning: no description for @param
[ERROR] * @param listener
[ERROR] ^
[ERROR] C:\git\DataCleaner\api\src\main\java\org\datacleaner\api\AnalyzerResultFuture.java:43: warning: no description for @param
[ERROR] * @param <R>
[ERROR] ^
```

When I execute `mvn site` it goes into an endless loop.

--- a/api/src/main/java/org/datacleaner/descriptors/DescriptorProvider.java
+++ b/api/src/main/java/org/datacleaner/descriptors/DescriptorProvider.java
@@ -52,9 +52,7 @@ public interface DescriptorProvider {
     public Collection<DescriptorProviderStatus> getStatus();
 
     /**
-     * It refreshes the descriptor list.
-     *
-     * @return
+     * Refreshes the descriptor list.
      */
     public void refresh();
 

--- a/pom.xml
+++ b/pom.xml
@@ -5,7 +5,7 @@
 		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
 
 		<!-- Maven plugin versions -->
-		<javadoc.version>2.9.1</javadoc.version>
+		<javadoc.version>2.10.3</javadoc.version>
 
 		<!-- Dependency versions -->
 		<scala.version>2.11.4</scala.version>

